be/src/io/shared_buffered_input_stream.h

```diff
@@ -118,6 +125,9 @@ class SharedBufferedInputStream : public SeekableInputStream {
     int64_t _direct_io_timer = 0;
     int64_t _align_size = 0;
     int64_t _estimated_mem_usage = 0;
+
+    // indicates the number of objects hold the shared_buffered_stream
+    std::atomic<int32_t> hold_count = 1;
```

Comment:
You're already using shared_ptr to manage InputStream, why do you need a separate atomic variable to manage reference counting? There should be a better way to do it